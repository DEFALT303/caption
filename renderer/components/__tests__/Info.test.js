import ReactTestRenderer from "react-test-renderer";
import Info from "./../Info";

describe("<Info />", () => {
  it("should show an empty state when no results are found", () => {
    const tree = ReactTestRenderer.create(<Info loading={false} results={[]} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should show the total results if any are present", () => {
    const results = [{}, {}];
    const tree = ReactTestRenderer.create(<Info loading={false} results={results} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should show a loader when there are no results yet", () => {
    const tree = ReactTestRenderer.create(<Info loading results={[]} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
