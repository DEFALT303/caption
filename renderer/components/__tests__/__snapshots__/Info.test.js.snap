// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Info /> should show a loader when there are no results yet 1`] = `
<div
  className="jsx-3536369025"
>
  <span
    className="jsx-904518150"
  >
    <img
      className="jsx-904518150"
      src="/static/loading.gif"
    />
  </span>
</div>
`;

exports[`<Info /> should show an empty state when no results are found 1`] = `
<div
  className="jsx-3536369025"
>
  <span
    className="jsx-3536369025"
  >
    Nothing Found
  </span>
</div>
`;

exports[`<Info /> should show the total results if any are present 1`] = `
<div
  className="jsx-3536369025"
>
  <span
    className="jsx-3536369025"
  >
    2 Results
  </span>
</div>
`;
